### PR TITLE
Unsafe

### PR DESCRIPTION
**Pull request by havardh**

### CODE REVIEW

File: background/rust.tex [hunk 1]
```diff
@@ -637,7 +637,7 @@ \subsection{Unsafe Code} % (fold)
 Sometimes, this means that the underlying code needs direct memory access and the ability to dereference raw pointers or call into external libraries.
 For these tasks, {\rust} provides the keyword {\unsafe}.
 It is important to note that this {\unsafe} keyword is introduced to \emph{keep} {\rust}'s semantics about memory safety.
-When the compiler can no longer guarantee safety, it is up to the programmer to tell the compiler that the code inside the {\unsafe} block is indeed safe.
+The programmer is responsible for maintaining the safe invariants of the compiler when using a {\unsafe} block.
```
Comment (sondrele) [inline background/rust.tex]: "using `an` unsafe block."

File: background/rust.tex [hunk 2]
```diff
@@ -669,13 +669,16 @@ \subsection{Unsafe Code} % (fold)
 The borrow-checker can never be disabled, but for certain operations it will simply not work.
 \autoref{lst:unsafe_reference} demonstrates this with the same example that was used in \autoref{ssec:lifetimes}.
 We can see from this example that it is not allowed to return a reference to something that is allocated \emph{within} an unsafe block, as this can result in use-after-free issues.
-This also tells us that using an unsafe block is not the same as disabling the compiler, it is merely a mechanism for the programmer to handle the safety of a program in the cases where the compiler is unable to.
+This also tells us that using an unsafe block is not the same as disabling the compiler, it is merely a mechanism for the programmer to take responsibility for the safety of part of the program in the cases where the compiler is unable to.
```
Comment (sondrele) [inline background/rust.tex]: hva med:
"This also tells us that using an {\unsafe} block is not the same as disabling the compiler, it is merely a mechanism for the programmer to `take responsibility for safety in the parts of the program where the compiler is unable to.`"

Litt rar setning generelt, men tror det blir bedre

File: background/rust.tex [hunk 3]
```diff
@@ -669,13 +669,16 @@ \subsection{Unsafe Code} % (fold)
...
 
 \begin{listing}[H]
 \begin{rustcode}
 fn main() {
   let name = unsafe {
     let b = Book { name: "Gecko's".to_string(), pages: 150 };
+    // If the last statement in a block does not contain a `;'
+    // then the block is turned into a expression which
```
Comment (sondrele) [inline background/rust.tex]: // then the block is turned into `an` expression `that`

Tror det er bedre med that :P

File: background/rust.tex [hunk 4]
```diff
@@ -669,13 +669,16 @@ \subsection{Unsafe Code} % (fold)
...
+    // evaluates to value of this statement
```
Comment (sondrele) [inline background/rust.tex]: // evaluates to `the` value of this statement

File: background/rust.tex (hunk 2 above)
Comment (havardh) [inline background/rust.tex]: Hva med: 
"This also tells us that using an {\unsafe} block is not the same as disabling `all the checks of` the compiler, it is merely a mechanism for the programmer to take responsibility for safety in the parts of the program where the compiler is unable to."

Det er rart å si slå av kompilatoren :P

Comment (sondrele) [inline background/rust.tex]: Ja.. :P det jeg egentlig mente å rette på var det som var etter "it is
merely..."!

**Pull request merged**